casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -44,6 +51,41 @@ object ConflictSetMerger {
         .getOrElse(Set.empty)
     }
 
+    def calMergedResult(
+        branch: R,
+        originResult: Map[Blake2b256Hash, Long]
+    ): Map[Blake2b256Hash, Long] = {
+      val diff = mergeableChannels.apply(branch)
+      diff.foldLeft(originResult) {
+        case (ba, br) =>
+          val result = Math.addExact(ba.getOrElse(br._1, 0L), br._2)
+          if (result < 0) {
+            throw new ArithmeticException("merged result negative")
```

Comment:
I am in a dilemma whether I should introduce effect F here to introduce `ApplicativeError` and raiseError or just throw error.

Reply:
We don't want to add more WartRemover annotations and use them only to existing code.

Because this error is a _platform fatal_ error you can use assertions which will throw exceptions.

Reply:
why do you consider that as `platform fatal error`? I am very curious.

Reply:
It's fatal in a sense that is not recoverable. Node must exit and not handle the error.

Reply:
Oh, I see you are handle the error to detect negative or overflow. Let's better to encode this in result type and return Option or Either.